### PR TITLE
Setup py3 test job

### PR DESCRIPTION
**Pull request by mvdbeek**
@jmchilton if the Dockerfile looks OK to you, could you build a new image ?

xref. #1715

### CODE REVIEW

File: test/docker/base/Dockerfile [hunk 1]
```diff
@@ -79,11 +81,15 @@ RUN mkdir /etc/galaxy && cd /tmp/ansible && mkdir roles && \
...
+        export GALAXY_VIRTUAL_ENV=$VENV && \
+        ./scripts/common_startup.sh || { echo "common_startup.sh failed"; exit 1; } && \
+        dev_requirements=./lib/galaxy/dependencies/dev-requirements.txt && \
+        [ -f $dev_requirements ] && $VENV/bin/pip install -r $dev_requirements; done
+
+RUN for VENV in $GALAXY_VIRTUAL_ENV_2 $GALAXY_VIRTUAL_ENV_3; do \
+    export GALAXY_VIRTUAL_ENV=$VENV && \
+    . $GALAXY_VIRTUAL_ENV/bin/activate && \
     pip install psycopg2 && \
     pip install mysql && \
```
Comment (jmchilton) [inline test/docker/base/Dockerfile]: The loop runs over the rest of this twice but only pre-populates the databases once right? This is fine I'm just double checking - it'd be a little cleaner to end the loop sooner if that is the case but we can merge this as is and do that in a subsequent revision - I know it takes a while to revise these things.

**Comment by jmchilton** [comment]
Yeah - I can build and publish a new image.

File: test/docker/base/Dockerfile (hunk 1 above)
Comment (mvdbeek) [reply test/docker/base/Dockerfile]: Good catch, it does call the script twice, I'll change that

**Comment by mvdbeek** [comment]
I'm a bit stuck on this now, the docker image doesn't build because `bdbag` pins `bagit==1.6.4`, and `bagit==1.6.4` can't be installed on python 3.5 -- which has been fixed in bagit 1.7.0:
```
(galaxy_venv3) root@e5e36d3697a2:/# pip install bagit
Requirement already satisfied: bagit in /galaxy_venv3/lib/python3.6/site-packages (1.7.0)
(galaxy_venv3) root@e5e36d3697a2:/# pip install bdbag
Collecting bdbag
  Downloading https://files.pythonhosted.org/packages/53/c1/c28ec526b699813287f94cd4b9971918881505ecb1e7760a3a0fa8d9bf9c/bdbag-1.4.1-py2.py3-none-any.whl (63kB)
    100% |################################| 71kB 838kB/s
Collecting certifi (from bdbag)
  Using cached https://files.pythonhosted.org/packages/7c/e6/92ad559b7192d846975fc916b65f667c7b8c3a32bea7372340bfe9a15fa5/certifi-2018.4.16-py2.py3-none-any.whl
Collecting bagit==1.6.4 (from bdbag)
  Using cached https://files.pythonhosted.org/packages/64/91/4363e2d56a81431ef7b86cd8f0a530a5729032f207eb3b756eb29b4ed282/bagit-1.6.4.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-k6o0335t/bagit/setup.py", line 19, in <module>
        long_description = readme.read()
      File "/galaxy_venv3/lib/python3.6/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1111: ordinal not in range(128)
```

**Comment by mvdbeek** [comment]
PR to fix this is [here](https://github.com/fair-research/bdbag/pull/28)

**Comment by jmchilton** [comment]
If we have a strict pin on a newer version - does everything fail at that point - I suppose it would? We can make bdbag an optional dependency until it is well behaved with respect to its dependencies in order to make progress. We can just ``@skip`` the relevant tests under Python 3.

**Comment by mvdbeek** [comment]
Should be straightforward, we can download the package and do `python setup.py install` until we have a new fixed version. Or is there a way to override the pin in bdbag's setup.py ?

**Comment by mvdbeek** [comment]
On the brightside the pytest setup works (in this instance we tested python 2): https://jenkins.galaxyproject.org/job/docker-api-py3/6/testReport/.

**Comment by mvdbeek** [comment]
Alright, this is passing now: https://jenkins.galaxyproject.org/job/docker-api-py3/10/
Before merging we'll need a new image and flip back the image to galaxy/testing-base:18.09.0

**Comment by jmchilton** [comment]
```
% docker pull 'mvdbeek/testing-base:18.09.0'
18.09.0: Pulling from mvdbeek/testing-base
8ee29e426c26: Pull complete
6e83b260b73b: Pull complete
e26b65fd1143: Pull complete
40dca07f8222: Pull complete
b420ae9e10b3: Pull complete
28dfb9ba5664: Pull complete
930784f0d6fc: Pull complete
1228d52f23a5: Pull complete
8c1f55b45f64: Pull complete
4de36f30db26: Pull complete
c52b1ac944b1: Pull complete
fb32b7d9670c: Pull complete
27ce2b3a0804: Pull complete
df4dde9b6ecd: Pull complete
525367ae10f8: Pull complete
4e60ccbb9a10: Pull complete
10f9fabb48c2: Pull complete
860fd37df1af: Pull complete
d35d59a11e9f: Pull complete
99646ea51eeb: Pull complete
42960f87951d: Pull complete
75378962b5d9: Pull complete
Digest: sha256:e34f9906cdaed7ec28d9627b0e165ba8312f014a355068f14406b6dccb4a1e07
Status: Downloaded newer image for mvdbeek/testing-base:18.09.0
% docker tag 'mvdbeek/testing-base:18.09.0' 'galaxy/testing-base:18.09.0'
% docker push 'galaxy/testing-base:18.09.0'
The push refers to repository [docker.io/galaxy/testing-base]
f4e6c61120a1: Mounted from mvdbeek/testing-base
1c4f959cba00: Mounted from mvdbeek/testing-base
b9ea605c7e6a: Mounted from mvdbeek/testing-base
875ebe897543: Mounted from mvdbeek/testing-base
2c31e2d6b7ec: Mounted from mvdbeek/testing-base
1b7d90784c1a: Mounted from mvdbeek/testing-base
baea7298b045: Mounted from mvdbeek/testing-base
e9d02a2a710a: Mounted from mvdbeek/testing-base
948192940c16: Mounted from mvdbeek/testing-base
f046bf1cba59: Mounted from mvdbeek/testing-base
65b401a9b1c0: Mounted from mvdbeek/testing-base
5da4616b40eb: Mounted from mvdbeek/testing-base
832b541b4b05: Mounted from mvdbeek/testing-base
9517e241e25a: Mounted from mvdbeek/testing-base
e2d2d156739f: Mounted from mvdbeek/testing-base
22ee48af0819: Mounted from mvdbeek/testing-base
5c269795d1e4: Mounted from mvdbeek/testing-base
bcff331e13e3: Mounted from mvdbeek/testing-base
2166dba7c95b: Mounted from mvdbeek/testing-base
5e95929b2798: Mounted from mvdbeek/testing-base
c2af38e6b250: Mounted from mvdbeek/testing-base
0a42ee6ceccb: Mounted from mvdbeek/testing-base
18.09.0: digest: sha256:e34f9906cdaed7ec28d9627b0e165ba8312f014a355068f14406b6dccb4a1e07 size: 4922
```

**Comment by mvdbeek** [comment]
Alright, integration tests (including TS installs, all thanks to @nsoranzo who moved the cloning to `hg` calls) should be running as well.

**Review by jmchilton — APPROVED** [review]

**Comment by jmchilton** [comment]
Wow - way to keep up the awesome. This is great! I guess we should wait for the integration tests to pass on Jenkins now that the mule tests are re-enabled - it was working before when running the tests all together but the mule tests would work when they were the only thing running.

**Comment by mvdbeek** [comment]
Yeah, I haven't set up the jenkins job since it's down, but it passes inside docker.

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
Had to wrestle a little more with `localhost` inside docker, but https://github.com/galaxyproject/galaxy/pull/6565/commits/dd7e4c601a3f31ce9c4fa06afc8234a39f0a5804 solved that. Here's a run of the integration test that passed on py3: 
https://jenkins.galaxyproject.org/job/integration-py3/3/testReport/

**Comment by jmchilton** [comment]
Amazing!

**Pull request merged**